multiversx/mx-chain-go
# fix gas used and fee relayed v3

## Reasoning behind the pull request
- Fixed how is calculated `gasUsed` and `fee` fields for `relayed v3` transactions  
## Proposed changes
- Implemented new functions that will calculate the gas used and fee filed for the API transactions and for the indexer transactions
- Changed the version of the `mx-chain-core-go` : https://github.com/multiversx/mx-chain-core-go/pull/330
- Changed the version of the `mx-chain-es-indexer-go`: https://github.com/multiversx/mx-chain-es-indexer-go/pull/298
## Testing procedure
- Do a system test and check if the gas used and the fee is set correctly for the API transactions and the indexed transactions

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
